### PR TITLE
Check incoming authed SAP in signed votes

### PR DESCRIPTION
**Pull request by grumbach**
- Make sure incoming vote's auth'd SAPs are correctly signed by elders before we forward them to consensus.

### CODE REVIEW

**Comment by grumbach** [comment]
Seems like the test `split_network_assert_health_check ` is failing on the latest main branch as well (without the changes included in this PR)

**Comment by grumbach** [comment]
> Seems like the test `split_network_assert_health_check ` is failing on the latest main branch as well (without the changes included in this PR)

Now should be fixed

File: sn_node/src/node/core/messaging/handling/handover.rs [hunk 1]
```diff
@@ -54,6 +95,12 @@ impl Node {
             signed_vote
         );
 
+        if let Err(err) = self.check_signed_vote_saps(&signed_vote) {
+            // NB TODO tracking who sent this invalid vote?
+            error!("Ignoring invalid handover vote: {}", err);
```
Comment (joshuef) [inline sn_node/src/node/core/messaging/handling/handover.rs:100]: Let's return an error here. I think we can then look to hook into dysfunction in a layer above.

**Review by joshuef — CHANGES_REQUESTED** [review]
Return an error in order to react above to dysfunction there.

**Review by joshuef — APPROVED** [review]

**Pull request merged**